test/parallel/test-child-process-argument-errors.js

```diff
@@ -0,0 +1,275 @@
+// Copyright Joyent, Inc. and other Node contributors.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a
+// copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to permit
+// persons to whom the Software is furnished to do so, subject to the
+// following conditions:
+//
+// The above copyright notice and this permission notice shall be included
+// in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+// OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN
+// NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+// OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE
+// USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const { spawn, fork, execFile } = require('child_process');
+const fixtures = require('../common/fixtures');
+
+const empty = fixtures.path('empty.js');
+const cmd = common.isWindows ? 'rundll32' : 'ls';
+
+// Argument types for combinatorics.
+const a = [];
+const o = {};
+function c() {}
+const s = 'string';
+const u = undefined;
+const n = null;
```

Comment:
Yes, but those were already there so probably deserve another PR to refactor..

Reply:
Hm, this is detected as a new file. So every line is considered "new". I therefore feel like it would be best to change that here, but I do not feel strongly about it.

Reply:
@BridgeAR Well..not really, since this file still has the license header, I don't think I can remove them simply because the file is refactored and renamed...(or CAN I?)

Reply:
Well... I am not a lawyer and I brought the same question up recently somewhere else.

Here in this case I feel it might actually be safe to remove it because it really has nothing to do with the original code that was used when publishing the file: https://github.com/nodejs/node/blame/f3473d7db696f78b284c7a874ba33ec5e2b91309/test/simple/test-child-process-spawn-typeerror.js

The reason why I feel like it is safe here is that I do not see any connection from the original code to the current one.

Reply:
@BridgeAR Other than the `invalidSignatureError` (~50% of this test), another ~50% code here only got moved around, indented and put into a block..I don't feel very safe in this case.

Reply:
Changing the code is always fine. So the change itself can be done.

And the question to what the license applies: I think to the very original code? In that case I do not see any connection from the current state to that code.